src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs

```diff
@@ -1626,16 +1626,9 @@ public TBServer()
 
     public void Dispose()
     {
-        try
-        {
-            if (process != null && !process.HasExited)
```

Comment:
@batiati process can't be null here, right? If we are running Dispose, constructor has finished?

Or is there some special C# semantics when Dispose might get called if you only managed to run through half of the constructor's code?

Reply:
Nice catch! Since `process` is readonly, it cannot be null after the constructor
https://github.com/tigerbeetle/tigerbeetle/blob/2cdabe44b44a54d279e2866da6812e1201506eb5/src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs#L1624

And objects that are not fully constructed don't run the finalizer/destructor/dispose.

I suppose the original intent was to prevent double `Dispose(),` but it's completely out of scope to cover this edge case in a test runner!